src/main/java/dev/blaauwendraad/masker/json/SegmentReference.java

```diff
@@ -0,0 +1,20 @@
+package dev.blaauwendraad.masker.json;
+
+/**
+ * A mutable reference to a sequence of bytes in <code>message</code>. It is used to represent json path segments.
+ * A "key" segment type reference is represented as a (start, offset) pair.
+ * For an "array index" segment type reference, a (start, offset) pair is interpreted as (index, -1).
+ */
+public class SegmentReference {
+    int start;
+    int offset;
```

Comment:
Is it for performance to not make there private and use getters? I think that will be optimized by Hotspot anyway

Reply:
Agree. I will make it a POJO bean.